### ISSUE TITLE
Document update conflicts when clearing schedules

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#6624]
**Describe the bug**
`accept_patient_reports` and `registration` fail with a conflict when clearing schedules. 

**To Reproduce**
1. Load `standard` config
2. Create a pregnancy (app form) for a patient. Wait for Sentinel to process this change.
3. Create a delivery (app form) for the same patient. Wait for Sentinel to process the change. 
4. Check Sentinel logs and see conflict. 
We also have a race condition where, if Sentinel's queue is reset quickly enough, all will seem well in the app - with the delivery doc updated successfully, even if there is a conflict the 1st time. Most of the time, the delivery doc will not be updated correctly. If multiple transitions run over it, you might even see an error on the doc stating that `registration` transition failed with a conflict. 

**Expected behavior**
Neither of these transitions should fail with a conflict. 

**Logs**
You will see something like this:
```
2020-08-31 15:43:17 ERROR: transition registration errored on doc <delivery doc ID> seq <SEQ>: {
  id: '<pregnancy doc ID>',
  error: 'conflict',
  reason: 'Document update conflict.',
  name: 'conflict',
  status: 409,
  message: 'Document update conflict.',
  docId: '<pregnancy doc ID>'
} 
```

Notice that the conflict isn't on the currently processed delivery doc, rather on the old pregnancy doc, whose schedules need clearing.

**Environment**
- Instance: local
- App: api, sentinel
- Version: 3.9, 3.10

**Additional context**
This problem appeared when `medic-client/reports_by_subject` was changed to emit multiple times per doc: https://github.com/medic/cht-core/commit/f31350c1f5e4dd792a71fc414c731807b1d3e2b2#diff-e503e25acb27edd4d19d7f0badcfedbbR9
This made the original `pregancy` show up multiple times in the results list and fail the subsequent "updates". 

The error also happens when processing SMSs, but if checking SMS, look at API logs instead of Sentinel logs.

**Comment by garethbowen** [comment]
Ready for AT.

**Comment by newtewt** [comment]
Tested this fix locally. Creating multiple docs with schedules that get cleared by another doc. There is no document conflict occurring anymore. Feel free to close and merge @dianabarsan

**Comment by dianabarsan** [comment]
Merged into `master` and backported to `3.10.x`.
